app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    class UsersController < API::V1::ApplicationController
+      include API::V1::ErrorHandlerConcern
+
+      skip_before_action :doorkeeper_authorize!, only: :create
+
+      def create
+        user = User.new(create_params.except(:client_id))
+        client_app = Doorkeeper::Application.find_by(uid: create_params[:client_id])
+
+        unless client_app
+          render_error 'Invalid client ID', status: :forbidden, source: :client_id
+          return
+        end
+
+        render_errors ActiveModel::ErrorsSerializer.new(user.errors).serializable_hash and return unless user.save
+
+        # create access token for the user, so the user won't need to login again after registration
+        access_token = user.create_access_token client_app.id
+        # return json containing access token and refresh token
+        # so that user won't need to call login API right after registration
+        render json: create_success_result(user, access_token), status: :created
```

Comment:
I think we need to use user serialized here similar to Ankit comment [here](https://github.com/malparty/google-search-ruby/pull/35/files#r654332811)
```ruby
  # this is pseudo code
  render json: UserSerializer.new(user)
```
We mainly render json response using serializer. Like we did in our previous projects([Servey](https://github.com/nimblehq/nimble-survey-web/blob/development/app/controllers/api/v1/surveys_controller.rb#L11), [The-1](https://github.com/nimblehq/the1-api/blob/development/engines/the1_idm/app/controllers/the1_idm/v2/accounts_controller.rb#L18))

What do you think?

Reply:
yes, cf other thread ;-)